src/database/hooks/useSaveAnonChatHook.ts

```diff
@@ -53,46 +65,35 @@ const useSaveAnonChatHook = () => {
 
     initAnonymousChat.message.cid = initAnonymousChat.message.cid.replace('messaging:', '');
 
-    const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
-    await chat.save(localDb);
-
     try {
+      const chat = ChatSchema.fromInitAnonymousChatAPI(initAnonymousChat, status);
+      await chat.save(localDb);
       const members = originalMembers || object?.members;
-      members?.forEach((member) => {
-        const saveUserAndChannelListMember = async () => {
-          try {
-            const userMember = UserSchema.fromInitAnonymousChatAPI(
-              member,
-              initAnonymousChat?.message?.cid
-            );
-
-            await userMember.saveOrUpdateIfExists(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile userMember', e);
-          }
-          try {
-            const memberSchema = ChannelListMemberSchema.fromInitAnonymousChatAPI(
-              initAnonymousChat?.message?.cid,
-              uuid(),
-              member
-            );
-            memberSchema.saveIfNotExist(localDb);
-          } catch (e) {
-            console.log('error saveChatFromOtherProfile memberSchema', e);
-          }
-        };
-
-        Promise.all([saveUserAndChannelListMember()]);
+      const promises: Promise<void>[] = members?.map((member) => {
+        return saveUserAndChannelListMember(initAnonymousChat.message.cid, member);
       });
+
+      await Promise.all(promises);
     } catch (e) {
-      console.log('error saveChatFromOtherProfile');
       console.log(e);
     }
-
-    refresh('channelList');
-    refresh('chat');
-
-    if (withNavigate) helperGoToAnonymousChat(object, GROUP_INFO);
+    try {
+      const channelListData = ChannelListSchema.fromInitAnonymousChatAPI(
+        {
+          ...initAnonymousChat
+        },
+        type === SIGNED ? 'PM' : 'ANON_PM'
+      );
+      await channelListData.saveIfLatest(localDb);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      refresh('channelList');
+      refresh('chat');
+      if (withNavigate) {
+        helperGoToAnonymousChat(object, GROUP_INFO);
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, it's good to see error handling being added. However, simply logging the error and continuing with the execution might lead to unexpected behavior. Consider rethrowing the error after logging it. Also, using `console.log(e)` for logging errors is not a best practice. Instead, use `console.error(e)`. 

```diff
-       console.log(e);
+       console.error(e);
+       throw e;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->